libraries/aws_iam_user.rb

```diff
@@ -15,29 +15,31 @@ class AwsIamUser < Inspec.resource(1)
   def initialize(
     opts,
     aws_user_provider = AwsIam::UserProvider.new,
+    aws_user_details_provider_ini = AwsIam::UserDetailsProviderInitializer.new,
     access_key_factory = AwsIamAccessKeyFactory.new
   )
-    @user = opts[:user]
-    @user = aws_user_provider.user(opts[:name]) if @user.nil?
+    user = opts[:user]
+    user = aws_user_provider.user(opts[:name]) if user.nil?
```

Comment:
I propose that we remove UserProvider: it's just a thin wrapper around the connection, at this point.

Reply:
Agreed. We can take care of that in an upcoming PR.

Reply:
I think that's fair.  This *is* an *interim* change on the way to lazy loading during filtering...